src/pybind/mgr/dashboard/frontend/src/app/shared/services/modal.service.ts

```diff
@@ -0,0 +1,28 @@
+import { Injectable } from '@angular/core';
+
+import { NgbModal, NgbModalOptions, NgbModalRef } from '@ng-bootstrap/ng-bootstrap';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ModalService {
+  constructor(private modal: NgbModal) {}
+
+  show(component: any, initialState?: any, options?: NgbModalOptions): NgbModalRef {
```

Comment:
Could you please add at least a test for show?

Reply:
Done